dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ParamTypesResolver.java

```diff
@@ -115,25 +116,32 @@ public Optional<ParameterType> getParameterType(EndpointName endpointName, Varia
         }
 
         // TODO(12345): More validation of values.
-
         AnnotationReflector annotationReflector = ImmutableAnnotationReflector.of(
                 Preconditions.checkNotNull(Iterables.getOnlyElement(paramAnnotationMirrors), "paramAnnotationMirrors"));
         if (annotationReflector.isAnnotation(Request.Body.class)) {
             // default annotation param values are not available at annotation processing time
             String serializerName = InstanceVariables.joinCamelCase(endpointName.get(), "Serializer");
             Optional<TypeMirror> customSerializer = annotationReflector.getValueFieldMaybe(TypeMirror.class);
             TypeMirror serializer = customSerializer.orElseGet(() -> context.getTypeMirror(Json.class));
+            TypeName serializerType = TypeName.get(serializer);
             if (context.isAssignable(variableElement.asType(), BinaryRequestBody.class)
                     && context.isSameTypes(serializer, Json.class)) {
                 context.reportError(
                         "BinaryRequestBody is not supported by default, prefer the more expressive RequestBody type",
                         variableElement);
                 return Optional.empty();
             }
-
+            CodeBlock serializerFactory;
+            if (customSerializer.isEmpty()) {
+                serializerFactory = CodeBlock.of("new $T(runtime.bodySerDe())", serializerType);
```

Comment:
nit: Same here, can we add a `Json(ConjureRuntime)` constructor overload?


```suggestion
                serializerFactory = CodeBlock.of("new $T(runtime)", serializerType);
```